src/main/java/com/thealgorithms/datastructures/hashmap/hashing/HashMapCuckooHashing.java

```diff
@@ -0,0 +1,256 @@
+package com.thealgorithms.datastructures.hashmap.hashing;
+
+
+import java.lang.Math;
+
+/**
+ * This class is an implementation of a hash table using Cuckoo Hashing It uses
+ * // * a dynamic array to lengthen the size of the hash table when load factor > .7
+ *
+ * https://en.wikipedia.org/wiki/Cuckoo_hashing
+ */
+public class HashMapCuckooHashing {
+
+    private int hsize; // size of the hash table
+    private Integer[] buckets; // array representing the table
+    private final Integer AVAILABLE;
+    private int size; // amount of elements in the hash table
+
+    private int thresh; // threshold for infinite loop checking
+
+    /**
+     * Constructor initializes buckets array, hsize, and creates dummy object
+     * for AVAILABLE
+     *
+     * @param hsize the desired size of the hash map
+     */
+    public HashMapCuckooHashing(int hsize) {
+        this.buckets = new Integer[hsize];
+        this.hsize = hsize;
+        this.AVAILABLE = Integer.MIN_VALUE;
+        this.size = 0;
+        this.thresh = (int) (Math.log(hsize) / Math.log(2)) + 2;
+    }
+
+    /**
+     * The 2 Hash Functions takes a given key and finds an index based on its data, 2 distinctive ways to minimize collisions
+     *
+     * @param key the desired key to be converted
+     * @return int an index corresponding to the key
+     */
+
+    public int hashing1(int key) {
+        int hash = key % hsize;
+        if (hash < 0) {
+            hash += hsize;
+        }
+        return hash;
+    }
+
+    public int hashing2(int key) {
+        int hash = key / hsize;
+        hash %= hsize;
+        if (hash < 0) {
+            hash += hsize;
+        }
+        return hash;
+    }
+
+    /**
+     * inserts the key into the hash map by wrapping it as an Integer object, then uses while loop to insert new key
+     * if desired place is empty, return.
+     * if already occupied, continue while loop over the new key that has just been pushed out.
+     * if while loop continues more than Thresh, rehash table to new size, then push again.
+     *
+     * @param key the desired key to be inserted in the hash map
+     */
+
+    public void insertHash(int key) {
+        Integer wrappedInt = key, temp;
+        int hash, maxloop = 0;
+
+        if (isFull()) {
+            System.out.println("Hash table is full\n");
```

Comment:
Please throw exceptions instead of writing to stdout

Reply:
Wouldn't throwing exception halt the program unnecessarily? 

Also, could you provide a good example from the repo for exception throwing?

Reply:
Unnecessarily? If you don't throw and don't insert the key into the table, you violate the assumption that the `insertHash` function inserts hash. And this comment applies throughout the class where preconditions for calling a method are not satisfied, or the provided argument is invalid, or something else goes wrong.
https://github.com/TheAlgorithms/Java/blob/9fb3364cccc5373c32dbdb661f773c5ee753c77a/src/main/java/com/thealgorithms/audiofilters/IIRFilter.java